src/enums/types.rs

```diff
@@ -0,0 +1,254 @@
+use serde::{Deserialize, Serialize};
+use std::error;
+use std::fmt;
+use std::str::FromStr;
+
+#[derive(Clone, Debug)]
+pub struct Error {
+    kind: ErrorKind,
+}
+
+impl Error {
```

Comment:
Couldn't this Error be removed and merge the ErrorKind with ClientError? Something like `ParseEnum` would be fine IMO.